.github/workflows/ci-aarchxx.yml

```diff
@@ -32,8 +32,12 @@
 
 name: ci-aarchxx
 on:
-  # Run on pushes to any repository branch.
+  # Run on pull requests and pushes to master.
+  # We do not run on pushes to feature branches: pull requests cover those.
   push:
+    branches:
+      - master
+  pull_request:
```

Comment:
The `pull_request` event is triggered by various sub-events; perhaps we don't need all of them? https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request

Also, could you try pushing a commit to this PR's branch, to check whether new pushes to existing PR indeed trigger the CI? (it could be covered under the `edited` event, but not sure)

Reply:
I looked at those docs while creating this change and it was a little confusing.  It says "By default, a workflow only runs when a pull_request's activity type is opened, synchronize, or reopened. To trigger workflows for more activity types, use the types keyword."  But it doesn't clearly define the types: it points at the REST docs which also does not define them.  I was hoping the default would match what both Travis and Appveyor have, i.e., creation plus pushes, but maybe not.

Searching others shows that "opened, synchronize, and edited" are a common combo:

https://github.com/search?q=pull_request+extension%3A.yml+path%3A.github%2Fworkflows&type=Code

So maybe "opened" means the initial creation, "synchronize" means update by merging in master, and "edited" means a push?  Posting this response here does not seem to count as "edited".

Reply:
I think that's what we want: the default plus "edited".  It did trigger on my new push.  I suppose with this level of control we could remove "edited" and have a trigger on all pushes to all branches if we want to run it on non-pull-request branches but I'm not sure that's what we want: and that would not be in sync with Appveyor.  Maybe once we've removed Appveyor if we switch to GA for Windows we can revisit at that point; for now I think it's best to mirror what we had before.